etc/templates/openapi3.tmpl

```diff
@@ -0,0 +1,346 @@
+                "tags": [
+                    "{{ schema.Metadata.resource_group }}"
+                ],{% endif %}
+                "requestBody": {
+                    "description": "{{ schema.ID }} resource input",
+                    "required": true,
+                    "content": {
+                        "application/json": {
+                            "schema": {
+                                "$ref": "#/components/schemas/{{ schema.ID }}Input"
+                            }
+                        }
+                    }
+                },
+                "responses": {
+                    "201": {
+                        "description": "{{ schema.ID }} created",
+                        "content": {
+                            "application/json": {
+                                "schema": {
+                                    "$ref": "#/components/schemas/{{ schema.ID }}"
+                                }
+                            }
+                        }
+                    },
+                    "default": {
+                        "$ref": "#/components/responses/UnexpectedError"
+                    }
+                }
+            }{% endif %}
+        },
+        "{{ schema.URL }}/{id}": {
+            {% if "read" in policies %}"get": {
+                "description" : "Show a {{ schema.ID}} resources",
+                {% if schema.Metadata.resource_group | length != 0 %}
+                "tags": [
+                    "{{ schema.Metadata.resource_group }}"
+                ],{% endif %}
+                "parameters": [
+                {
+                    "name": "id",
+                    "in": "path",
+                    "description": "ID of {{ schema.ID }} to fetch",
+                    "required": true,
+                    "schema": {
+                        "type": "string"
+                    }
+                }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "{{ schema.Description }}",
+                        "content": {
+                            "application/json": {
+                                "schema": {
+                                    "$ref": "#/components/schemas/{{ schema.ID }}"
+                                }
+                            }
+                        }
+                    },
+                    "default": {
+                        "$ref": "#/components/responses/UnexpectedError"
+                    }
+                }
+            }{% if "update" in policies or "delete" in policies %},{% endif %}{% endif %}
+            {% if "update" in policies %}"put": {
+                "description" : "Update {{ schema.ID }} resource",
+                {% if schema.Metadata.resource_group | length != 0 %}
+                "tags": [
+                    "{{ schema.Metadata.resource_group }}"
+                ],{% endif %}
+                "parameters": [
```

Comment:
"put" requires both "id" and "body"

Reply:
Nice catch, fixed and added test for `requestBody`.